src/types/seeds.ts

```diff
@@ -0,0 +1,119 @@
+
+import { Table } from '../db/db';
+import db from '../db/sql-db'
+
+import { ArtistProfileKeys } from './artist';
+import { getCrdtUpdateMessage, getCrdtUpsertMessage } from './message'
+import { NFTContractTypeName, NFTFactoryKeys, NFTStandard } from './nft';
+import { MusicPlatform, MusicPlatformKeys, MusicPlatformType } from './platform';
+import { ProcessedTrackKeys } from './track';
+
+enum SeedEntities {
+  'platforms',
+  'nftFactories',
+  'artistProfiles',
+  'processedTracks'
+}
+type SeedEntity = keyof typeof SeedEntities;
+
+enum PlatformsRequiredKeys {
+  ID = 'id',
+  NAME = 'name',
+  TYPE = 'type',
+}
+
+enum NFTFactoriesRequiredKeys {
+  ID = 'id',
+  PLATFORM_ID = 'platformId',
+  CONTRACT_TYPE = 'contractType', // validated against `NFTContractTypeName`
+  STANDARD = 'standard', // validated against `NFTStandard`
+  AUTO_APPROVE = 'autoApprove',
+  APPROVED = 'approved'
+}
+
+enum ArtistProfilesRequiredKeys {
+  ARTIST_ID = 'artistId',
+  PLATFORM_ID = 'platformId',
+}
+
+enum ProcessedTracksRequiredKeys {
+  ARTIST_ID = 'artistId',
+  PLATFORM_ID = 'platformId',
+}
+
+type SeedPayload = {
+  entity: SeedEntity,
+  data: any
+}
+
+export const validateSeed = (payload: SeedPayload): void => {
+  entityValidator(payload);
+
+  const validatorFunctions = {
+    'platforms': [
+      () => minimumKeysPresent(payload, Object.values(PlatformsRequiredKeys)),
+      () => onlyValidKeysPresent(payload, MusicPlatformKeys),
+      () => typeValidator(payload, 'type', MusicPlatformType),
+    ],
+    'nftFactories': [
+      () => minimumKeysPresent(payload, Object.values(NFTFactoriesRequiredKeys)),
+      () => onlyValidKeysPresent(payload, NFTFactoryKeys),
+      () => typeValidator(payload, 'contractType', NFTContractTypeName),
+      () => typeValidator(payload, 'standard', NFTStandard),
+    ],
+    'artistProfiles': [
+      () => minimumKeysPresent(payload, Object.values(ArtistProfilesRequiredKeys)),
+      () => onlyValidKeysPresent(payload, ArtistProfileKeys),
+    ],
+    'processedTracks': [
+      () => minimumKeysPresent(payload, Object.values(ProcessedTracksRequiredKeys)),
+      () => onlyValidKeysPresent(payload, ProcessedTrackKeys),
```

Comment:
`minimumKeysPresent` and `onlyValidKeysPresent` are used in each one of the supported seed entities. I imagine they'll get replaced with a more robust runtime TS/Schema validator library in future.

Reply:
ye seems fine